### PR TITLE
Moddataimp 116 - Update fileDefinition status on upload fail

### PR DESCRIPTION
**Pull request by RuslanLavrov**

### CODE REVIEW

File: src/main/java/org/folio/service/processing/ParallelFileChunkingProcessor.java [hunk 1]
```diff
@@ -90,19 +89,11 @@ public void process(JsonObject jsonRequest, JsonObject jsonParams) { //NOSONAR
...
-                    .map(e -> uploadDefinitionService.updateBlocking(uploadDefinition.getId(), definition -> Future.succeededFuture(uploadDefinition.withStatus(COMPLETED)), tenantId))
-                    .otherwise(uploadDefinitionService.updateBlocking(uploadDefinition.getId(), definition -> Future.succeededFuture(uploadDefinition.withStatus(UploadDefinition.Status.ERROR)), tenantId));
-                  }
-                }
+                null
               );
             }
             return Future.succeededFuture();
-          });
+          }).setHandler(event -> uploadDefinitionService.updateBlocking(uploadDefinition.getId(), definition -> Future.succeededFuture(definition.withStatus(COMPLETED)), tenantId));
```
Comment (OleksiiKuzminov) [inline src/main/java/org/folio/service/processing/ParallelFileChunkingProcessor.java]: format code plz to more readable view

File: src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java [hunk 2]
```diff
@@ -378,6 +380,16 @@ private Errors validateFreeSpace(UploadDefinition definition, Errors errors) {
     }, tenantId);
   }
 
+  @Override
+  public Future<Boolean> updateUploadDefinitionStatusToError(UploadDefinition uploadDefinition, String tenantId) {
```
Comment (OleksiiKuzminov) [inline src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java]: bad practice. Use status as a method argument

**Comment by Igor-Gorchakov** [comment]
Please update news.md

File: src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java [hunk 3]
```diff
@@ -378,6 +380,16 @@ private Errors validateFreeSpace(UploadDefinition definition, Errors errors) {
...
+  @Override
+  public Future<Boolean> updateUploadDefinitionStatusIfAllFilesUploadingFailed(UploadDefinition uploadDefinition, UploadDefinition.Status status, String tenantId) {
+    return uploadDefinition.getFileDefinitions()
+      .stream()
+      .allMatch(fileDefinition -> fileDefinition.getStatus().equals(ERROR))
+        ? uploadDefinitionDao.updateBlocking(uploadDefinition.getId(), definition ->
+          Future.succeededFuture(definition.withStatus(UploadDefinition.Status.ERROR)), tenantId)
+          .map(true)
+        : Future.succeededFuture(false);
+  }
```
Comment (Anatolii-Starkov) [inline src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java]: @RuslanLavrov Don't go crazy. I can't read it 😄 
I suggest the following
```suggestion
 public Future<Boolean> updateUploadDefinitionStatusIfAllFilesUploadingFailed(
          UploadDefinition uploadDefinition,
          Status status,
          String tenantId) {

    Future<Boolean> result = Future.succeededFuture(false);

    if (areAllDefinitionsFailed(uploadDefinition)) {
      result = uploadDefinitionDao
              .updateBlocking(uploadDefinition.getId(), it -> Future.succeededFuture(it.withStatus(Status.ERROR)), tenantId)
              .map(true);
    }

    return result;
  }


  private boolean areAllDefinitionsFailed(UploadDefinition uploadDefinition) {

    return uploadDefinition.getFileDefinitions().stream()
            .allMatch(it -> it.getStatus() == FileDefinition.Status.ERROR);
  }
```

File: src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java [hunk 4]
```diff
@@ -378,6 +380,22 @@ private Errors validateFreeSpace(UploadDefinition definition, Errors errors) {
     }, tenantId);
   }
 
+  @Override
+  public Future<Boolean> updateUploadDefinitionStatusIfAllFilesUploadingFailed(UploadDefinition uploadDefinition, UploadDefinition.Status status, String tenantId) {
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java]: do we really need UploadDefinition.Status status argument for this method?

Comment (RuslanLavrov) [reply src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java]: at first the argument did not exist, but the second comment from @OleksiiKuzminov says that it's bad practice

Comment (OleksiiKuzminov) [reply src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java]: Can you just make method update status and call it if condition is true?

File: src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java [hunk 5]
```diff
@@ -378,6 +378,22 @@ private Errors validateFreeSpace(UploadDefinition definition, Errors errors) {
...
+    Future<Boolean> result = Future.succeededFuture(false);
+
+//    if(areAllDefinitionsFailed(uploadDefinition)) {
```
Comment (OleksiiKuzminov) [inline src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java]: ?

File: src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java [hunk 6]
```diff
@@ -378,6 +378,22 @@ private Errors validateFreeSpace(UploadDefinition definition, Errors errors) {
     }, tenantId);
   }
 
+  @Override
+  public Future<Boolean> updateUploadDefinitionStatusIfAllFilesUploadingFailed(UploadDefinition uploadDefinition, UploadDefinition.Status status, String tenantId) {
```
Comment (OleksiiKuzminov) [inline src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java]: we dont need this method

**Pull request merged**